src/GitHub.Exports/Helpers/ThreadingHelper.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Threading.Tasks;
+using ThreadHelper = Microsoft.VisualStudio.Shell.ThreadHelper;
+using GitHub.Extensions;
+using System.Runtime.CompilerServices;
+using System;
+using static Microsoft.VisualStudio.Threading.JoinableTaskFactory;
+using static Microsoft.VisualStudio.Threading.AwaitExtensions;
+
+namespace GitHub.Helpers
+{
+    public interface IAwaitable
+    {
+        IAwaiter GetAwaiter();
+    }
+
+    public interface IAwaiter : INotifyCompletion
+    {
+        bool IsCompleted { get; }
+        void GetResult();
+    }
+
+    public static class ThreadingHelper
+    {
+        /// <summary>
+        /// Switch to the UI thread using ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync
+        /// Auto-disables switching when running in unit test mode
+        /// </summary>
+        /// <returns></returns>
+        public static IAwaitable SwitchToMainThreadAsync()
+        {
+            return Guard.InUnitTestRunner() ?
```

Comment:
Yeah, I think this is me misunderstanding `ConfigureAwait` - I'd assumed that the `ConfigureAwait(false)` state would be inherited in the Task that was returned, but yeah I see now that it'd affect the task that's being awaited rather than the task being returned. Wonder if there's a way to `ConfigureAwait(false)` the resulting task?